### PR TITLE
Finalized persistence!

### PR DESCRIPTION
**Pull request by kester-ng**
- Finalized persistence
- Added type erasure for repoprotocol to allow adapter pattern

### CODE REVIEW

File: Challo/Challo/Persistence/Stores/Booking/BookingModelConvertor.swift [hunk 1]
```diff
@@ -0,0 +1,125 @@
+//
+//  BookingModelConvertor.swift
+//  Challo
+//
+//  Created by Kester Ng on 13/4/21.
+//
+
+/*
+class BookingModelConvertor {
```
Comment (TanLeYang) [inline Challo/Challo/Persistence/Stores/Booking/BookingModelConvertor.swift:9]: Can remove all these commented out stuff

File: Challo/Challo/Persistence/Stores/Guide/GuideModelConvertor.swift [hunk 2]
```diff
@@ -0,0 +1,121 @@
+//
+//  GuideModelConvertor.swift
+//  Challo
+//
+//  Created by Kester Ng on 13/4/21.
+//
+
+/*
+class GuideModelConvertor {
```
Comment (TanLeYang) [inline Challo/Challo/Persistence/Stores/Guide/GuideModelConvertor.swift:9]: Can remove all these commented out stuff

File: Challo/Challo/Persistence/Stores/Review/ReviewModelConvertor.swift [hunk 3]
```diff
@@ -0,0 +1,91 @@
+//
+//  ReviewModelConvertor.swift
+//  Challo
+//
+//  Created by Kester Ng on 13/4/21.
+//
+
+/*
+class ReviewModelConvertor {
```
Comment (TanLeYang) [inline Challo/Challo/Persistence/Stores/Review/ReviewModelConvertor.swift:9]: Can remove all these commented out stuff

**Review by TanLeYang — APPROVED** [review]
Just remove the commented out chunks then lgtm!

File: Challo/Challo/Persistence/Stores/Booking/BookingModelConvertor.swift (hunk 1 above)
Comment (kester-ng) [reply Challo/Challo/Persistence/Stores/Booking/BookingModelConvertor.swift:9]: i forgot to remove, okok

**Pull request merged**